File: frontend/src/types/inquilino.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct Inquilino {
    pub id: String,
    pub nombre: String,
    pub apellido: String,
    pub email: Option<String>,
    pub telefono: Option<String>,
    pub cedula: String,
    pub contacto_emergencia: Option<String>,
    pub notas: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct CreateInquilino {
    pub nombre: String,
    pub apellido: String,
    pub email: Option<String>,
    pub telefono: Option<String>,
    pub cedula: String,
    pub contacto_emergencia: Option<String>,
    pub notas: Option<String>,
}

#[derive(Debug, Clone, Serialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct UpdateInquilino {
    pub nombre: Option<String>,
    pub apellido: Option<String>,
    pub email: Option<String>,
    pub telefono: Option<String>,
    pub cedula: Option<String>,
    pub contacto_emergencia: Option<String>,
    pub notas: Option<String>,
}

File: android/core/model/src/main/kotlin/com/propmanager/core/model/Inquilino.kt
package com.propmanager.core.model

import java.time.Instant

data class Inquilino(
    val id: String,
    val nombre: String,
    val apellido: String,
    val email: String?,
    val telefono: String?,
    val cedula: String,
    val contactoEmergencia: String?,
    val notas: String?,
    val createdAt: Instant,
    val updatedAt: Instant,
    val isPendingSync: Boolean = false,
)
